src/api/internal.rs

```diff
@@ -619,7 +619,7 @@ impl<T: Pixel> ContextInner<T> {
               TileSuperBlockOffset(SuperBlockOffset { x: sbx, y: sby });
             let half_res_pmvs =
               build_half_res_pmvs(fi, ts, tile_sbo, &tile_pmvs);
-            ts.half_res_pmvs.push(half_res_pmvs);
+            ts.half_res_pmvs[sbx + sby * ts.sb_width] = half_res_pmvs;
```

Comment:
To keep this safer would be a good idea to use the rayon collect capabilities to save the half_res_pmvs outside of the parallel loop.